data/sql_setup/prepare_schedule_a.sql

```diff
@@ -14,6 +25,54 @@ create index on ofec_sched_a_queue_old (timestamp);
 create index on ofec_sched_a_queue_new (two_year_transaction_period);
 create index on ofec_sched_a_queue_old (two_year_transaction_period);
 
+
+-- Support for processing of schedule A itemized records that need to be
+-- retried.
+create or replace function retry_processing_schedule_a_records(start_year integer) returns void as $$
+declare
+    timestamp timestamp = current_timestamp;
+    two_year_transaction_period smallint;
+    view_row fec_vsum_sched_a_vw%ROWTYPE;
+    schedule_a_record record;
+begin
+    for schedule_a_record in select * from ofec_sched_a_nightly_retries loop
+        select into view_row * from fec_vsum_sched_a_vw where sub_id = schedule_a_record.sub_id;
+
+        if found then
+            two_year_transaction_period = get_transaction_year(view_row.contb_receipt_dt, view_row.rpt_yr);
+
+            if two_year_transaction_period >= start_year then
+                -- Determine which queue(s) the found record should go into.
+                case schedule_a_record.action
+                    when 'insert' then
+                        delete from ofec_sched_a_queue_new where sub_id = view_row.sub_id;
+                        insert into ofec_sched_a_queue_new values (view_row.*, timestamp, two_year_transaction_period);
+
+                        delete from ofec_sched_a_nightly_retries where sub_id = schedule_a_record.sub_id;
+                    when 'delete' then
```

Comment:
Do we want to get rid of the view check and delete the record directly by sub id? That can be a follow up task if we want to get this moving sooner